Makefile

```diff
@@ -1,106 +1,62 @@
-KUBEWARDEN_DEFAULTS_CHART_OLD_VERSION   := $(shell echo '$(KW_VERSIONS)' | jq -er '.old["kubewarden/kubewarden-defaults"]' || echo "*")
-
-# CRD version to be tested
-CRD_VERSION ?= $(shell helm show values $(KUBEWARDEN_CHARTS_LOCATION)/kubewarden-defaults --version '$(KUBEWARDEN_DEFAULTS_CHART_VERSION)' | yq ".crdVersion")
-CRD_VERSION_SUFFIX ?= $(shell echo $(CRD_VERSION) | cut -d'/' -f2)
-# directory with all the files used during the tests. This files are copied from
-# $(ROOT_RESOURCES_DIR) and changed to used the CRDs version defined in $(CRD_VERSION)
-RESOURCES_DIR ?= $(ROOT_RESOURCES_DIR)/resources_$(CRD_VERSION_SUFFIX)
+MKFILE_DIR ?= $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+TESTS_DIR ?= $(MKFILE_DIR)tests
+RESOURCES_DIR ?= $(MKFILE_DIR)resources
 
+NAMESPACE ?= kubewarden
+CLUSTER_CONTEXT ?= $(shell kubectl config current-context)
 
 # ==================================================================================================
-# Aliases
-kube = kubectl --context $(CLUSTER_CONTEXT) $(1)
-helm = helm --kube-context $(CLUSTER_CONTEXT) $(1)
-bats = RESOURCES_DIR=$(RESOURCES_DIR) \
-	KUBEWARDEN_CRDS_CHART_OLD_VERSION=$(KUBEWARDEN_CRDS_CHART_OLD_VERSION) \
-	KUBEWARDEN_DEFAULTS_CHART_OLD_VERSION=$(KUBEWARDEN_DEFAULTS_CHART_OLD_VERSION) \
-	KUBEWARDEN_CONTROLLER_CHART_OLD_VERSION=$(KUBEWARDEN_CONTROLLER_CHART_OLD_VERSION) \
-	KUBEWARDEN_CRDS_CHART_VERSION=$(KUBEWARDEN_CRDS_CHART_VERSION) \
-	KUBEWARDEN_DEFAULTS_CHART_VERSION=$(KUBEWARDEN_DEFAULTS_CHART_VERSION) \
-	KUBEWARDEN_CONTROLLER_CHART_VERSION=$(KUBEWARDEN_CONTROLLER_CHART_VERSION) \
-	KUBEWARDEN_CHARTS_LOCATION=$(KUBEWARDEN_CHARTS_LOCATION) \
-	KUBEWARDEN_HELM_REPO_NAME=$(KUBEWARDEN_HELM_REPO_NAME) \
-	CLUSTER_CONTEXT=$(CLUSTER_CONTEXT) \
-	NAMESPACE=$(NAMESPACE) \
-		bats -T --print-output-on-failure $(1)
+# Optional arguments for scripts
 
-helm_in = $(helm) upgrade --install --wait --namespace $(NAMESPACE) --create-namespace
+# cluster_k3d.sh:
+#   K3S=[1.30] - short|long version
+#   CLUSTER_NAME=[k3d-default]
 
-# ==================================================================================================
-# Macros
-define install-kubewarden =
-	$(helm_in) kubewarden-crds $(KUBEWARDEN_CHARTS_LOCATION)/kubewarden-crds --version "$(KUBEWARDEN_CRDS_CHART_VERSION)"
-	$(helm_in) kubewarden-controller $(KUBEWARDEN_CHARTS_LOCATION)/kubewarden-controller --version "$(KUBEWARDEN_CONTROLLER_CHART_VERSION)"
-	$(helm_in) kubewarden-defaults $(KUBEWARDEN_CHARTS_LOCATION)/kubewarden-defaults --version "$(KUBEWARDEN_DEFAULTS_CHART_VERSION)"
-	$(kube) wait --for=condition=Ready --namespace $(NAMESPACE) pods --all
-endef
-
-define generate-versioned-resources-dir =
-	./scripts/generate_resources_dir.sh $(ROOT_RESOURCES_DIR) $(CRD_VERSION)
-endef
+# helmer.sh:
+#   VERSION=[next|prev|v1.17.0-rc2|local] (app version)
+#   REPO_NAME=[kubewarden]
+#   CHARTS_LOCATION=[./dirname|reponame]
+#   LATEST=[1]
+#   CRDS_ARGS, DEFAULTS_ARGS, CONTROLLER_ARGS
```

Comment:
Can we remove this comments?

Reply:
I think these are intended to work as inline docs. I'm fine with them.

Reply:
yes, it was meant as inline docs, maybe it would be better as README
I would leave it there and maybe move in next iteration..